core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -176,18 +179,17 @@ class LoadBalancerService(
         entry
     }
 
-    /**
-     * When invoker health detects a new invoker has come up, this callback is called.
-     */
+    /** When invoker health detects that an invoker has gone unhealthy or offline, this callback is called. */
     private def clearInvokerState(invokerName: String) = {
-        val actSet = activationByInvoker.getOrElseUpdate(invokerName, new TrieSet[ActivationEntry])
-        actSet.keySet map {
-            case actEntry @ ActivationEntry(activationId, namespaceId, invokerIndex, _, promise) =>
-                promise.tryFailure(new LoadBalancerException(s"Invoker $invokerIndex restarted"))
-                activationById.remove(activationId)
-                activationByNamespaceId.get(namespaceId) map { _.remove(actEntry) }
+        activationByInvoker.get(invokerName).foreach { actSet =>
+            actSet.keySet map {
+                case actEntry @ ActivationEntry(activationId, namespaceId, invokerIndex, _, promise) =>
```

Comment:
Is this active ack promise termination necessary? What if an invoker comes back online and the requests were in the queue anyway. The recovery mechanism on the active ack will clean up the active acks if it's really catastrophic but may take longer.